src/platform_linux/platform.c

```diff
@@ -103,16 +111,21 @@ platform_buffer_getaddr(const buffer_handle *bh)
    return bh->addr;
 }
 
+/*
+ * platform_buffer_destroy() - Destroy the large buffer created using mmap()
+ * Free all associated memory. buffer_handle *bh_in is NULL'ed out on exit.
+ */
 platform_status
-platform_buffer_destroy(buffer_handle *bh)
+platform_buffer_destroy(platform_heap_id heap_id, buffer_handle **bh_in)
```

Comment:
Changed the interface to receive address of `buffer_handle *`, so that it can be NULL'ed out on exit.
(This is a safer way to avoid callers accidentally accessing `bh_in` after this call.